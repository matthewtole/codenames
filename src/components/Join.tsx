import * as React from 'react';

export interface JoinProps {
  error?: string;
  onSubmit: (code: string) => void;
}

interface State {
  code: string;
}

const VALID_CODE = /^[0-9]{6}$/;

export class Join extends React.Component<JoinProps, State> {
  constructor(props: JoinProps) {
    super(props);

    this.state = {
      code: '',
    };
  }

  render() {
    return (
      <div className="box">
        <form onSubmit={this.handleSubmit}>
          {this.props.error && (
            <div className="notification is-danger">{this.props.error}</div>
          )}
          <div className="control">
            <input
              className="input has-text-centered is-large"
              type="tel"
              placeholder="Room Code"
              value={this.state.code}
              onChange={this.handleCodeChange}
            />
          </div>
          <br />
          <div className="has-text-centered">
            <input
              type="submit"
              disabled={!this.isValidCode}
              className="button is-success is-medium"
              onClick={this.handleSubmit}
              style={{ width: '100%' }}
              value="Join Game"
            />
          </div>
        </form>
      </div>
    );
  }

  private get isValidCode(): boolean {
    return VALID_CODE.test(this.state.code);
  }

  private handleSubmit = (
    event: React.MouseEvent<HTMLInputElement> | React.FormEvent<HTMLFormElement>
  ) => {
    if (event) {
      event.stopPropagation();
      event.preventDefault();
    }
    if (!this.isValidCode) {
      return;
    }

    this.props.onSubmit(this.state.code);
  }

  private handleCodeChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    this.setState({
      code: event.target.value,
    });
  }
}
