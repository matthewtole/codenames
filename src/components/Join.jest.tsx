import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Join, JoinProps } from './Join';

describe('<Join />', () => {
  const defaultProps: JoinProps = {
    onSubmit: jest.fn(),
  };

  it('renders correctly', () => {
    const tree = renderer.create(<Join {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
