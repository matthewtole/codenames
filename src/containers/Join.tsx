import { connect, Dispatch } from 'react-redux';
import { State } from '../redux/store';
import { Action } from 'redux';
import { joinRoom } from '../redux/room/action_creators';
import { Join } from '../components/Join';

interface JoinProps {}

interface JoinStateProps {
  error?: string;
}

interface JoinDispatchProps {
  onSubmit: (code: string) => void;
}

type Props = JoinProps & JoinDispatchProps & JoinStateProps;

const mapStateToProps = (state: State, ownProps: Props): JoinStateProps => {
  return {
    error: state.room.error,
  };
};

const mapDispatchToProps = (
  dispatch: Dispatch<Action>,
  ownProps: Props
): JoinDispatchProps => {
  return {
    onSubmit: (code: string) => dispatch(joinRoom({ code })),
  };
};

export const JoinContainer = connect<
  JoinStateProps,
  JoinDispatchProps,
  JoinProps
>(mapStateToProps, mapDispatchToProps)(Join);
