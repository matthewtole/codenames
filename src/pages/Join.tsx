import * as React from 'react';
import { JoinContainer } from '../containers/Join';
import { Navbar } from '../components/Navbar';

export class JoinPage extends React.Component<{}, {}> {
  render() {
    return (
      <div className="has-navbar-fixed-top">
        <Navbar page="create" />
        <JoinContainer />
      </div>
    );
  }
}
