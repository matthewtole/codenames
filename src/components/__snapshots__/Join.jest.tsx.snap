// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Join /> renders correctly 1`] = `
<div>
  <section
    className="hero is-primary"
  >
    <div
      className="hero-body"
    >
      <div
        className="container has-text-centered"
      >
        <h1
          className="title"
        >
          Join Existing Game
        </h1>
      </div>
    </div>
  </section>
  <section
    className="section"
  >
    <div
      className="container is-fluid"
    >
      <div
        className="columns is-mobile is-centered"
      >
        <div
          className="column is-half-desktop is-two-thirds-tablet is-full-mobile is-one-third-widescreen "
        >
          <div
            className="box"
          >
            <div
              className="control"
            >
              <input
                className="input has-text-centered is-large"
                onChange={[Function]}
                placeholder="Room Code"
                type="text"
                value=""
              />
            </div>
            <br />
            <div
              className="has-text-centered"
            >
              <button
                className="button is-primary is-medium"
                disabled={true}
                onClick={[Function]}
                style={
                  Object {
                    "width": "100%",
                  }
                }
              >
                Join Game
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
